inference-engine/src/mkldnn_plugin/nodes/mkldnn_def_conv_node.h

```diff
@@ -86,29 +86,84 @@ class MKLDNNDeformableConvolutionNode : public MKLDNNNode {
     InferenceEngine::Precision getRuntimePrecision() const override;
 
 private:
-    size_t group = 1;
-    bool with_bilinear_pad = false;
-    std::vector<ptrdiff_t> stride = {};
-    std::vector<ptrdiff_t> dilation = {};
-    std::vector<ptrdiff_t> paddingL = {};
-
-    int deformable_group = 1;
-
-    jit_def_conv_params jcp = {};
-
-    std::vector<int> sampledCoordsVector;
-    std::vector<float> interpWeightsVector;
-
-    std::shared_ptr<jit_uni_def_conv_kernel> def_conv_kernel = nullptr;
-
-    void prepareSamplingWeights(const std::vector<size_t>& src_strides, const float* offsets, const std::vector<size_t>& off_strides,
-                                const float* modulation = nullptr, const std::vector<size_t>& modulation_strides = {});
-
-    void executeReference(const float* src, const float* weights, float* dst, const std::vector<size_t>& src_strides,
-                          const std::vector<size_t>& wei_strides, const std::vector<size_t>& dst_strides);
-    void executeOptimized(const float* src, const float* weights, float* dst,
-                          const std::vector<size_t>& src_strides, const std::vector<size_t>& dst_strides);
+    struct DefConvAttr {
+        size_t group = 1;
+        int deformable_group = 1;
+        bool with_bilinear_pad = false;
+        std::vector<ptrdiff_t> stride = {};
+        std::vector<ptrdiff_t> dilation = {};
+        std::vector<ptrdiff_t> paddingL = {};
+        std::vector<ptrdiff_t> paddingR = {};
+    } defConvAttr;
+
+    void prepareParams() override;
+    void updatePadding();
+    void executeDynamicImpl(mkldnn::stream strm) override;
+    static constexpr size_t DATA_ID = 0;
+    static constexpr size_t OFF_ID = 1;
+    static constexpr size_t WEI_ID = 2;
+    static constexpr size_t MOD_ID = 3;
+    std::string errorPrefix;
+    class DefConvExecutor {
+    public:
+        DefConvExecutor(const DefConvAttr &defConvAttr,
+                            const VectorDims &srcDims,
+                            const VectorDims &weiDims,
+                            const VectorDims &dstDims,
+                            const bool with_modulation);
+
+        jit_def_conv_params jcp = {};
+
+        virtual void exec(const float* src, const float* weights, float* dst, const std::vector<size_t>& src_strides,
```

Comment:
I don't see the essential difference. We can as much include in ctor off_strides and modulation_strides to prevent using of them in the previous line. Then we'll have ctor with 14 args. Why we pass anything in exec() then? I think more logical to pass all arguments, received from getParentEdgeAt(...) in one place.

Reply:
I guess better to separate runtime parameters and parameter that can be obtained on compilation stage.
To reduce args number you can pass descriptors instead passing dims and strides separately.

`exec` should take `src, weights, dst, offsets, modulation` 

Also `prepareSamplingWeights` should be called inside `exec`